src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -27,13 +27,125 @@
 #include <ripple/nodestore/impl/DecodedBlob.h>
 #include <ripple/nodestore/impl/EncodedBlob.h>
 #include <ripple/nodestore/impl/Shard.h>
+#include <boost/algorithm/hex.hpp>
 #include <chrono>
+#include <fstream>
+#include <iostream>
+#include <openssl/ripemd.h>
 #include <test/jtx.h>
 #include <test/nodestore/TestBase.h>
 
 namespace ripple {
 namespace NodeStore {
 
+/* std::uniform_int_distribution is platform dependent */
```

Comment:
Ugh... I don't know that I care for all this. I understand the need, given the platform-specific implementation details of `std::uniform_int_distribution`. I wonder if there's a better option here...

Reply:
What's the motivation for doing this? It it's to make sure we can repro unit test bugs on all platforms, then I'd just live with the platform dependence and remove this class. There is a very small possibility that we'll see failures on some platforms that we won't be able to repro on other platforms in the first place. If this happens, we'll just have to debug on the platforms where we can repro.

(if there's another reason for the class then nvm, I only took a very quick look before weighing in).

Reply:
My unit test for deterministic shards is the following: it generates predictable accounts and transactions, packs them into ledgers and makes the shard. The hash of this shard should be equal to the given value. I saw that on different platforms (precisely, linux and mac) hashes of the resulting shard was different. I unvestigated this case and determine that the problem is in the class std::uniform_int_distribution which generates different pseudorandom sequences on different platforms, but we need predictable sequence.

Reply:
@p2peer That is a different motivation that what I was envisioning. Thanks for the clarification!

Reply:
Done: added comment to the code